src/SourceLocator/Type/Composer/Factory/MakeLocatorForComposerJsonAndInstalledJson.php

```diff
@@ -38,19 +40,26 @@ public function __invoke(string $installationPath, Locator $astLocator): SourceL
             throw InvalidProjectDirectory::atPath($installationPath);
         }
 
-        $composerJsonPath  = $realInstallationPath . '/composer.json';
-        $installedJsonPath = $realInstallationPath . '/vendor/composer/installed.json';
+        $composerJsonPath = $realInstallationPath . '/composer.json';
 
         if (! file_exists($composerJsonPath)) {
             throw MissingComposerJson::inProjectPath($installationPath);
         }
 
+        /** @var array{
+         * autoload: array{classmap: array<int, string>, files: array<int, string>, psr-4: array<string, array<int, string>>, psr-0: array<string, array<int, string>>},
+         * config: array{vendor-dir?: string}
+         * }|null $composer
+         */
+        $composer  = json_decode((string) file_get_contents($composerJsonPath), true, 512, JSON_THROW_ON_ERROR);
```

Comment:
I think I have done the `JSON_THROW_ON_ERROR` wrong. The test `testWillFailToProduceLocatorForProjectWithInvalidComposerJson` fails now as the exection type isn't the one expected anymore. 

I didn't know the `JSON_THROW` you talked about but this was what found, so really not sure if that was what you meant @Ocramius?

Reply:
Oh, I thought `testWillFailToProduceLocatorForProjectWithInvalidComposerJson` was not handled at all. If we already have tests for that, I would not add `JSON_THROW_ON_ERROR` :-)